pkg/liqo-controller-manager/external-network/externalnetwork-controller/externalnetwork_controller.go

```diff
@@ -0,0 +1,414 @@
+			return err
+		}
+		return controllerutil.SetControllerReference(extNet, pubKey, r.Scheme)
+	}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *ExternalNetworkReconciler) ensureGatewayClient(ctx context.Context,
+	extNet *networkingv1alpha1.ExternalNetwork) (*networkingv1alpha1.GatewayClient, error) {
+	if extNet.Spec.ServerEndpoint == nil {
+		return nil, fmt.Errorf("no remote ServerEndpoint found for ExternalNetwork %q", extNet.Name)
+	}
+
+	var remoteClusterID string
+	if extNet.Labels != nil {
+		if v, ok := extNet.Labels[consts.ReplicationOriginLabel]; ok {
+			remoteClusterID = v
+		}
+	}
+	if remoteClusterID == "" {
+		return nil, fmt.Errorf("no remote ClusterID found for ExternalNetwork %q", extNet.Name)
+	}
+
+	opts := &gatewayclient.ForgeOptions{
+		KubeClient:        r.KubeClient,
+		RemoteClusterID:   remoteClusterID,
+		GatewayType:       gatewayclient.DefaultGatewayType,
+		TemplateName:      gatewayclient.DefaultTemplateName,
+		TemplateNamespace: r.LiqoNamespace,
+		MTU:               r.MTU,
+		Addresses:         extNet.Spec.ServerEndpoint.Addresses,
+		Port:              extNet.Spec.ServerEndpoint.Port,
+	}
+	if extNet.Spec.ServerEndpoint.Protocol != nil {
+		opts.Protocol = string(*extNet.Spec.ServerEndpoint.Protocol)
+	} else {
+		opts.Protocol = string(corev1.ProtocolTCP)
+	}
+	gwClient, err := gatewayclient.ForgeGatewayClient(getName(extNet, false), extNet.Namespace, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err = controllerutil.CreateOrUpdate(ctx, r.Client, gwClient, func() error {
+		if err = gatewayclient.MutateGatewayClient(gwClient, opts); err != nil {
+			return err
+		}
+		return controllerutil.SetControllerReference(extNet, gwClient, r.Scheme)
+	}); err != nil {
+		return nil, err
+	}
+	return gwClient, nil
+}
+
+func (r *ExternalNetworkReconciler) getServerEndpoint(ctx context.Context,
+	gw *networkingv1alpha1.GatewayServer) (*networkingv1alpha1.EndpointStatus, error) {
+	if err := r.Client.Get(ctx, client.ObjectKey{
+		Name:      gw.Name,
+		Namespace: gw.Namespace,
+	}, gw); err != nil {
+		return nil, err
+	}
+
+	ep := gw.Status.Endpoint
+	if ep == nil {
+		return nil, fmt.Errorf("no Endpoint found for GatewayServer %q", gw.Name)
+	}
+	return ep, nil
+}
+
+func (r *ExternalNetworkReconciler) ensureGatewayServer(ctx context.Context,
+	extNet *networkingv1alpha1.ExternalNetwork) (*networkingv1alpha1.GatewayServer, error) {
+	var remoteClusterID string
+	if extNet.Labels != nil {
+		if v, ok := extNet.Labels[consts.ReplicationDestinationLabel]; ok {
+			remoteClusterID = v
+		}
+	}
+	if remoteClusterID == "" {
+		return nil, fmt.Errorf("no remote ClusterID found for ExternalNetwork %q", extNet.Name)
+	}
+
+	opts := &gatewayserver.ForgeOptions{
+		KubeClient:        r.KubeClient,
+		RemoteClusterID:   remoteClusterID,
+		GatewayType:       gatewayserver.DefaultGatewayType,
+		TemplateName:      gatewayserver.DefaultTemplateName,
+		TemplateNamespace: r.LiqoNamespace,
```

Comment:
```suggestion
		TemplateNamespace: gatewayserver.DefaultTemplateNamespace,
```

Reply:
This value is received by a flag indicating where liqo is installed; I think it's safer than the variable